pkg/apis/pipeline/v1/taskrun_types.go

```diff
@@ -0,0 +1,473 @@
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TaskRunList contains a list of TaskRun
+type TaskRunList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []TaskRun `json:"items"`
+}
+
+// GetPipelineRunPVCName for taskrun gets pipelinerun
+func (tr *TaskRun) GetPipelineRunPVCName() string {
+	if tr == nil {
+		return ""
+	}
+	for _, ref := range tr.GetOwnerReferences() {
+		if ref.Kind == pipeline.PipelineRunControllerName {
+			return fmt.Sprintf("%s-pvc", ref.Name)
+		}
+	}
+	return ""
+}
+
+// HasPipelineRunOwnerReference returns true of TaskRun has
+// owner reference of type PipelineRun
+func (tr *TaskRun) HasPipelineRunOwnerReference() bool {
+	for _, ref := range tr.GetOwnerReferences() {
+		if ref.Kind == pipeline.PipelineRunControllerName {
+			return true
+		}
+	}
+	return false
+}
+
+// IsDone returns true if the TaskRun's status indicates that it is done.
+func (tr *TaskRun) IsDone() bool {
+	return !tr.Status.GetCondition(apis.ConditionSucceeded).IsUnknown()
+}
+
+// HasStarted function check whether taskrun has valid start time set in its status
+func (tr *TaskRun) HasStarted() bool {
+	return tr.Status.StartTime != nil && !tr.Status.StartTime.IsZero()
+}
+
+// IsSuccessful returns true if the TaskRun's status indicates that it is done.
+func (tr *TaskRun) IsSuccessful() bool {
+	return tr != nil && tr.Status.GetCondition(apis.ConditionSucceeded).IsTrue()
+}
+
+// IsCancelled returns true if the TaskRun's spec status is set to Cancelled state
+func (tr *TaskRun) IsCancelled() bool {
+	return tr.Spec.Status == TaskRunSpecStatusCancelled
+}
+
+// HasTimedOut returns true if the TaskRun runtime is beyond the allowed timeout
+func (tr *TaskRun) HasTimedOut(ctx context.Context, c clock.PassiveClock) bool {
+	if tr.Status.StartTime.IsZero() {
+		return false
+	}
+	timeout := tr.GetTimeout(ctx)
+	// If timeout is set to 0 or defaulted to 0, there is no timeout.
+	if timeout == apisconfig.NoTimeoutDuration {
+		return false
+	}
+	runtime := c.Since(tr.Status.StartTime.Time)
+	return runtime > timeout
+}
+
+// GetTimeout returns the timeout for the TaskRun, or the default if not specified
+func (tr *TaskRun) GetTimeout(ctx context.Context) time.Duration {
+	// Use the platform default is no timeout is set
+	if tr.Spec.Timeout == nil {
+		defaultTimeout := time.Duration(config.FromContextOrDefaults(ctx).Defaults.DefaultTimeoutMinutes)
+		return defaultTimeout * time.Minute
+	}
+	return tr.Spec.Timeout.Duration
+}
+
+// GetNamespacedName returns a k8s namespaced name that identifies this TaskRun
+func (tr *TaskRun) GetNamespacedName() types.NamespacedName {
+	return types.NamespacedName{Namespace: tr.Namespace, Name: tr.Name}
+}
+
+// IsPartOfPipeline return true if TaskRun is a part of a Pipeline.
+// It also return the name of Pipeline and PipelineRun
+func (tr *TaskRun) IsPartOfPipeline() (bool, string, string) {
```

Comment:
This isn't on you, but this function doesn't look like something we want to exist-- TaskRuns should in general not need to know whether they are part of a pipelinerun or not.

Reply:
Thanks Lee. Opened [#5320](https://github.com/tektoncd/pipeline/issues/5320) to track this down.